### PR TITLE
OpenBox.io

### PR DESCRIPTION
**Pull request by openbox-NFT**
### Project Abstract

> Web3.0 & Mystery Box and Multichain NFT Marketplace
>
>

### For which grant [level](https://github.com/w3f/Grants-Program#level_slider-levels) are you applying? 
- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $50,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for > $100k Web3 Foundation Council approval)

### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied, renamed ( `project_name.md`) and updated.
- [x] A BTC or Ethereum (DAI/USDT) address for the payment of the milestones is provided inside the application.
- [x] I have read and acknowledged the [terms and conditions](https://github.com/w3f/Grants-Program/blob/master/docs/T&Cs.md).
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process).

### How Did You Hear About our grants program?

- [x] Social Media
- [x] Hackathon
- [x] Personal Recommendation
- [ ] Substrate Builders Program
- [ ] Investor/VC
- [x] Online Search 
- [x] Other: I led many people to participate in DOT's first ICO round in 2017 and have been following the development of Polkadot and Web3Foundation.

### CODE REVIEW

**Review by takahser — CHANGES_REQUESTED** [review]
Hi @openbox-NFT 
Thanks for your application. Currently, the M1 is missing lots of details and mostly consists of example texts, while M2 is missing completely. In general, it's not clear to me what you're planning to deliver as part of this grant. Feel free to add more details and ping me.

**Comment by openbox-NFT** [comment]
> Hi @openbox-NFT Thanks for your application. Currently, the M1 is missing lots of details and mostly consists of example texts, while M2 is missing completely. In general, it's not clear to me what you're planning to deliver as part of this grant. Feel free to add more details and ping me.

Hi@[takahser](https://github.com/takahser)  I have done the modification of the details missing  parts

**Review by takahser — CHANGES_REQUESTED** [review]
@openbox-NFT thanks for the update, it looks a bit better. However, there are still many details missing. Especially in the milestones deliveries it would be better to include additional information on what you're planning to build.

Also, I have a few specific questions:

- How does M1.2 differ from the existing [FRAME EVM pallet](https://paritytech.github.io/frontier/frame/evm.html)?
- How does M1.3 differ from the [FRAME Uniques pallet](https://github.com/paritytech/substrate/tree/master/frame/uniques)? Also, are you aware of the [Enjin's efforts](https://efinity.io/whitepaper/efinity-parachain) to build their [paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)?
- M1.4 we generally do not fund deployment efforts. Instead we typically limit on funding the development of open-source components that can be reused by the community.
- M2.0d: How does it differ from M1.0d?

**Comment by openbox-NFT** [comment]
> @openbox-NFT感谢更新，看起来好多了。但是，仍然缺少许多细节。尤其是在里程碑交付中，最好包含有关您计划构建的附加信息。
> 
> 另外，我有几个具体问题：
> 
> * M1.2 与现有的[FRAME EVM 托盘](https://paritytech.github.io/frontier/frame/evm.html)有何不同？
> * M1.3 与[FRAME Uniques 托盘](https://github.com/paritytech/substrate/tree/master/frame/uniques)有何不同？另外，您知道Enjin为构建他们的[paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)[所做的努力](https://efinity.io/whitepaper/efinity-parachain)吗？[](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)
> * M1.4 我们通常不资助部署工作。相反，我们通常会限制为社区可重用的开源组件的开发提供资金。
> * M2.0d：它与 M1.0d 有何不同？

@takahser  Sorry, because of the "template expression" in the "application-template.md” file:
2. Substrate module: Y We will create a Substrate module that will...
3. Substrate module: Z We will create a Substrate module that will...
affected some of our expression results.

Modification has been completed

**Comment by openbox-NFT** [comment]
> @openbox-NFT thanks for the update, it looks a bit better. However, there are still many details missing. Especially in the milestones deliveries it would be better to include additional information on what you're planning to build.
> 
> Also, I have a few specific questions:
> 
> * How does M1.2 differ from the existing [FRAME EVM pallet](https://paritytech.github.io/frontier/frame/evm.html)?
> * How does M1.3 differ from the [FRAME Uniques pallet](https://github.com/paritytech/substrate/tree/master/frame/uniques)? Also, are you aware of the [Enjin's efforts](https://efinity.io/whitepaper/efinity-parachain) to build their [paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)?
> * M1.4 we generally do not fund deployment efforts. Instead we typically limit on funding the development of open-source components that can be reused by the community.
> * M2.0d: How does it differ from M1.0d?

@takahser Sorry, because of the "template expression" in the "application-template.md” file:
2. Substrate module: Y We will create a Substrate module that will...
3. Substrate module: Z We will create a Substrate module that will...
affected some of our expression results.

Modification has been completed

**Review by semuelle — CHANGES_REQUESTED** [review]
Hey @openbox-NFT. I just had a look at your application and found that all of the following sections still only contain template text. Please remove the template text and replace it with your own. These are important for us to evaluate a grant application. Thank you!

-  Ecosystem Fit
-  Development Status
- Development Roadmap

Also, the specification of your deliverables should be verifiable and as specific as possible. For example, "continue to expand to support" is too vague. You also haven't explained anywhere what "Mystery Box" is.

**Review by takahser — CHANGES_REQUESTED** [review]
@openbox-NFT if I'm not mistaken, you haven't replied to my questions in [my previous comment](https://github.com/w3f/Grants-Program/pull/922#pullrequestreview-964162085) yet. Feel free to take a look again and to elaborate on it.
Also, I agree with @semuelle's [comment](https://github.com/w3f/Grants-Program/pull/922#pullrequestreview-977272684), please have a look at it as well and feel free to update the application in accordance it it.

**Comment by openbox-NFT** [comment]
> @openbox-NFT感谢更新，看起来好多了。但是，仍然缺少许多细节。尤其是在里程碑交付中，最好包含有关您计划构建的附加信息。
> 
> 另外，我有几个具体问题：
> 
> * M1.2 与现有的[FRAME EVM 托盘](https://paritytech.github.io/frontier/frame/evm.html)有何不同？
> * M1.3 与[FRAME Uniques 托盘](https://github.com/paritytech/substrate/tree/master/frame/uniques)有何不同？另外，您知道Enjin为构建他们的[paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)[所做的努力](https://efinity.io/whitepaper/efinity-parachain)吗？[](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)
> * M1.4 我们通常不资助部署工作。相反，我们通常会限制为社区可以重用的开源组件的开发提供资金。
> * M2.0d：它与 M1.0d 有何不同？
It's just a demo problem caused by the template, just a misunderstanding


> @openbox-NFT thanks for the update, it looks a bit better. However, there are still many details missing. Especially in the milestones deliveries it would be better to include additional information on what you're planning to build.
> 
> Also, I have a few specific questions:
> 
> * How does M1.2 differ from the existing [FRAME EVM pallet](https://paritytech.github.io/frontier/frame/evm.html)?
> * How does M1.3 differ from the [FRAME Uniques pallet](https://github.com/paritytech/substrate/tree/master/frame/uniques)? Also, are you aware of the [Enjin's efforts](https://efinity.io/whitepaper/efinity-parachain) to build their [paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)?
> * M1.4 we generally do not fund deployment efforts. Instead we typically limit on funding the development of open-source components that can be reused by the community.
> * M2.0d: How does it differ from M1.0d?

It's just a demo problem caused by the template, just a misunderstanding.Relevant content has been adjusted

**Comment by openbox-NFT** [comment]
> 嘿@openbox-NFT. 我刚刚查看了您的应用程序，发现以下所有部分仍然只包含模板文本。请删除模板文本​​并将其替换为您自己的。这些对我们评估拨款申请很重要。谢谢！
> 
> * 生态系统契合度
> * 发展状况
> * 发展路线图
> 
> 此外，您的可交付成果的规范应该是可验证的并且尽可能具体。例如，“继续扩大支持”就太模糊了。你也没有在任何地方解释什么是“神秘盒子”。

**Comment by openbox-NFT** [comment]
> @openbox-NFT thanks for the update, it looks a bit better. However, there are still many details missing. Especially in the milestones deliveries it would be better to include additional information on what you're planning to build.
> 
> Also, I have a few specific questions:
> 
> * How does M1.2 differ from the existing [FRAME EVM pallet](https://paritytech.github.io/frontier/frame/evm.html)?
> * How does M1.3 differ from the [FRAME Uniques pallet](https://github.com/paritytech/substrate/tree/master/frame/uniques)? Also, are you aware of the [Enjin's efforts](https://efinity.io/whitepaper/efinity-parachain) to build their [paratoken](https://github.com/w3f/PSPs/blob/4dabb4111efb9983455f8649db8cabb4a73b9b3d/PSPs/drafts/psp-32.md)?
> * M1.4 we generally do not fund deployment efforts. Instead we typically limit on funding the development of open-source components that can be reused by the community.
> * M2.0d: How does it differ from M1.0d?

It's just a demo problem caused by the template, just a misunderstanding.Relevant content has been adjusted

**Comment by openbox-NFT** [comment]
> Hey @openbox-NFT. I just had a look at your application and found that all of the following sections still only contain template text. Please remove the template text and replace it with your own. These are important for us to evaluate a grant application. Thank you!
> 
> * Ecosystem Fit
> * Development Status
> * Development Roadmap
> 
> Also, the specification of your deliverables should be verifiable and as specific as possible. For example, "continue to expand to support" is too vague. You also haven't explained anywhere what "Mystery Box" is.

Thanks for the reminder; relevant content has been adjusted

**Comment by openbox-NFT** [comment]
> @openbox-NFT if I'm not mistaken, you haven't replied to my questions in [my previous comment](https://github.com/w3f/Grants-Program/pull/922#pullrequestreview-964162085) yet. Feel free to take a look again and to elaborate on it. Also, I agree with @semuelle's [comment](https://github.com/w3f/Grants-Program/pull/922#pullrequestreview-977272684), please have a look at it as well and feel free to update the application in accordance it it.

Sorry, it does take time to understand and adjust the relevant content. We thank you for your continued guidance. Maybe we have problems understanding the related content, which leads to many problems. But we wanted to be able to get funding, but the delivery was underprepared. The team has to solve development problems, funding problems, and marketing problems. There is also looking for capital to negotiate and solve the problem of Pre-Seed round financing. Many of our submissions may not be perfect. But we are not trying to create a fake program to defraud donor grants. We also do not want to raise funds from the public by issuing Tokens because, in the 2.0 stage, there are practical commercial applications that need to be implemented, so we hope that it will comply with the laws of some regions as much as possible. I helped many communities participate in the first round of ICOs in the first round of Polkadot in 2017. I also helped a project raise 5000 ETH without charging them any commissions and holding their tokens. I just thought it was funny and socially meaningful. We participated in the first round of ICO of Polkadot, and we are also a participant in the community. We hope to receive grants and wish to support the ecology of Polkadot, too. But the rules always need to be followed, and we will try our best to do what we can with time and energy

File: applications/OpenBox.md [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+          && item.state == State.Created
+          && (IERC721(item.nftContract).getApproved(item.tokenId) == address(this))
+        )? true
+         : false;
+      }else{
+        return false;
+      }
+      }
+
+      }
```
Comment (semuelle) [inline applications/OpenBox.md:344]: What are these showing?

**Review by semuelle — CHANGES_REQUESTED** [review]
Hey @openbox-NFT, thanks for the update. Looking at it now, it seems to be a better fit for the grants program of an EVM-compatible parachain, such as Moonbeam. We don't usually fund Solidity development, and it sounds like that is your main priority. You mentioned building on Substrate, but if you do, you will have to [provide a list of pallets and functions](https://github.com/w3f/Grants-Program/blob/master/docs/grant_guidelines_per_category.md#runtime-moduleschains) that you want to implement.

**Comment by Noc2** [comment]
I’m closing this application due to inactivity. Let me know if I should reopen it.

File: applications/OpenBox.md (hunk 1 above)
Comment (openbox-NFT) [reply applications/OpenBox.md:344]: This part is a contract based on Solidity's NFT MarketPlace

**Comment by openbox-NFT** [comment]
> I’m closing this application due to inactivity. Let me know if I should reopen it.

The development funding issue was resolved some time ago and has been driving the development of MVP products recently. So there is no time to process this commit. We did more detailed research on Substrate, MoonBeam, and Unique Network. To achieve our expected cross-chain verification to complete the transaction function of the NFT order contract on another, one is developing a new application chain to achieve cross-chain interoperability support involves need a long cycle, and there is no plan to issue tokens at this stage. The other uses the existing MoonBeam developed based on Substrate and the existing pallets and interfaces on Unique Network to achieve interoperability. Unique Network does not seem to be doing enough on Ethereum to support cross-chain interoperability. It is impossible to realize cross-chain contract purchases smoothly directly and may need to rely on centralized steps or centralized trading pools to achieve step by step. The specific implementation may need to be put at a later stage

> I’m closing this application due to inactivity. Let me know if I should reopen it.

**Comment by openbox-NFT** [comment]
> I’m closing this application due to inactivity. Let me know if I should reopen it.

Thank you for your patient guidance. We can now focus on driving the MVP release and validation market. Support for the Substrate and Polkadot ecosystems will come at a later stage. Under the background of sufficient funds and relatively mature technology.

File: applications/OpenBox.md [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+| 2. | Use Substrate  to be compatible with Solidity contracts | Develop NFT, minting, and trading contracts for BSC using Substrate FRAME EVM compatible with Solidity.And complete the NFT development support of ETH and Polygon that belong to EVM.|  
+| 3. | Backend | Complete the back-end user name, collection information management, data retrieval, NFT trade price tracking function development..., related API access debugging.  https://github.com/UniqueNetwork/unique-marketplace-api#readme |  | 
+| 4. | Test | Complete relevant tests of NFT MarketPlace functions developed with Substrate and generate test reports |  
+
+
+### Milestone 2 Example — Additional features
+
+- **Estimated Duration:** 1 month
+- **FTE:**  5
+- **Costs:** 20000 USD
```
[the remark this answers is not in the document]
Comment (openbox-NFT) [reply applications/OpenBox.md:437]: Here are some changes in compensation and benefits for probationary and regular employees

**Pull request closed without merging**